win32/src/winapi/gdi32/palette.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{winapi::{gdi32::{Object, HDC, HGDIOBJ}, types::WORD}, Machine};
+
+const TRACE_CONTEXT: &'static str = "gdi32/palette";
+
+#[repr(C)]
+#[derive(Clone, Debug)]
+pub struct LOGPALETTE_HEADER {
+    palVersion: WORD,
+    palNumEntries: WORD,
+}
+unsafe impl memory::Pod for LOGPALETTE_HEADER {}
+
+#[repr(C)]
+#[derive(Clone, Debug)]
+pub struct PALETTEENTRY {
+    pub peRed: u8,
+    pub peGreen: u8,
+    pub peBlue: u8,
+    pub peFlags: u8,
+}
+unsafe impl memory::Pod for PALETTEENTRY {}
+
+#[derive(Debug)]
+pub struct Palette(LOGPALETTE_HEADER, Vec<PALETTEENTRY>);
+
+#[win32_derive::dllexport]
+pub fn CreatePalette(machine: &mut Machine, plpal: u32) -> HGDIOBJ {
+    let header = machine.mem().view::<LOGPALETTE_HEADER>(plpal).clone();
```

Comment:
This is better as:

let header = machine.mem().get_pod::<LOGPALETTE_HEADER>(plpal);

(Ultimately I think I might need to remove the .view() methods, because Rust is unhappy if the underlying pointers aren't aligned)